src/server/notifications/WebHookSubscription2021/WebHookUnsubscriber.ts

```diff
@@ -0,0 +1,50 @@
+import type { CredentialsExtractor } from '../../../authentication/CredentialsExtractor';
+import { ResetResponseDescription } from '../../../http/output/response/ResetResponseDescription';
+import type { ResponseDescription } from '../../../http/output/response/ResponseDescription';
+import { getLoggerFor } from '../../../logging/LogUtil';
+import { ForbiddenHttpError } from '../../../util/errors/ForbiddenHttpError';
+import { NotFoundHttpError } from '../../../util/errors/NotFoundHttpError';
+import type { OperationHttpHandlerInput } from '../../OperationHttpHandler';
+import { OperationHttpHandler } from '../../OperationHttpHandler';
+import type { SubscriptionStorage } from '../SubscriptionStorage';
+import type { WebHookFeatures } from './WebHookSubscription2021';
+
+/**
+ * Allows clients to unsubscribe from a WebHookSubscription2021.
+ * Assumed the trailing part of the incoming URL is the identifier of the subscription.
```

Comment:
Isn't that, like, super dangerous?

Reply:
If not, tell us why we're allowed to make that assumption.

Reply:
Only the person that made the subscription should have access to that identifier (and thus the unsubscribe URL), we don't expose it anywhere. Could have generated a new value just for the unsubscribe URL but thought we could just reuse the ID. Currently the spec doesn't say the unsubscribe request should be authenticated: 

> If the client wishes to unsubscribe from the webhook, it can make a DELETE request to the provided `unsubscribe_endpoint`.

Reply:
Ah okay, so it is a server-generated identifier? That's important context. So we'd deed to write then indeed who is responsible for creating the URLs.

My preference is here that we encode that knowledge in one separate place. So have one class/module/utility (can be embedded in this file) which has `createWebHookUrl(id)` and `getWebHookId(url)` or whatever is appropriate. Or just two member methods that are next to each other. But should not be implicitly distributed.

Reply:
Yea, currently the `WebHookSubscription2021.ts`, that generates the response, has a `unsubscribe_endpoint: '${this.unsubscribePath}/${encodeURI(info.id)}'` line. Could have a `WebHookUnsubscribeGenerator` or something like that with a `get(id: string): string` and `parse(url: string): string` functions.

Note that the WebSockets solution had a similar case where one class generates the WebSocket URL, and another class parses the URL of a connection to extract the ID from it: 
https://github.com/CommunitySolidServer/CommunitySolidServer/blob/37ba4040583426f4e974058f5931bad62d082aec/src/server/notifications/WebSocketSubscription2021/WebSocketSubscription2021.ts#L51
https://github.com/CommunitySolidServer/CommunitySolidServer/blob/37ba4040583426f4e974058f5931bad62d082aec/src/server/notifications/WebSocketSubscription2021/WebSocket2021Listener.ts#L29

Reply:
Yeah even if just utility functions. And same for the sockets indeed.